spec/controllers/api/v1/patients_controller_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::PatientsController, type: :controller do
+  describe "POST sync" do
+
+    let(:patients) {{patients: FactoryBot.attributes_for_list(:patient, 10)}}
+    it "creates new patients" do
+      post :sync_from_user, params: patients
+      expect(Patient.count).to eq 10
+      expect(response).to have_http_status(200)
+    end
+
+    it "updates the existing patients" do
+      existing_patients = FactoryBot.create_list(:patient, 10)
+      updated_patients = existing_patients.take(5).each do |p|
+        FactoryBot.attributes_for(
```

Comment:
Shouldn't this actually update the record `p`? Also, please don't use one-letter variables; make it `patient` instead for readability.

Reply:
Hrm, ya. This might be our clojure/haskell practices rubbing off on us, will be wary. This is fixed though.